words/string.asm

```diff
@@ -265,59 +264,38 @@ z_minus_leading:
 xt_minus_trailing:
                 jsr underflow_2
 w_minus_trailing:
-                ; if length entry is zero, return a zero and leave the
-                ; address part untouched
+                ; if length is zero we-re done
                 lda 0,x         ; LSB of n
                 ora 1,x         ; MSB of n
                 beq _done
 
-                ; Compute address of last char in tmp1 as
-                ; addr + u1 - 1
-
-                ; addr + u1
-                clc
-                lda 2,x         ; LSB of addr
-                adc 0,x
-                sta tmp1
-                lda 3,x         ; MSB of addr
-                adc 1,x
-                sta tmp1+1
-
-                ; - 1
-                lda tmp1
-                bne +
-                dec tmp1+1
-+
-                dec tmp1
+                ; Compute address past last character: addr + u1
+                jsr w_two_dup
+                jsr w_plus
+                ; ( addr u addr' )
 
 _loop:
-                ; While spaces are found, move tmp1 backwards and
-                ; decrease the count on the data stack.
-                lda (tmp1)
-                cmp #AscSP
-                bne _done
+                ; Move back to point at last character
+                jsr w_one_minus
 
-                ; Move back one address.
-                lda tmp1
-                bne +
-                dec tmp1+1
-+
-                dec tmp1
+                ; While spaces are found,
+                ; decrease the count on the data stack and repeat
+                lda (0,x)
+                jsr is_whitespace
```

Comment:
this also slows it down, but seems better than just cmp #' ' unless that was intentionally avoiding other whitespace?